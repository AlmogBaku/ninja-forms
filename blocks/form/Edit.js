import React from "react";
import { SelectControl, Placeholder } from "@wordpress/components";
import { InspectorControls } from "@wordpress/block-editor";
import useForms from "./useForms";
import FormPreviewIFrame from "./FormPreviewIFrame";
import { NinjaIcon } from "./icon";
import PropTypes from "prop-types";

const BlockPlaceholder = props => {
	return (
		<Placeholder
			icon={<div style={{ marginRight: "10px" }}>{NinjaIcon}</div>}
			label="Ninja Form"
			instructions="Display a form"
			style={{ alignItems: "center" }}
		>
			<div
				style={{
					margin: "auto",
					width: "50%"
				}}
			>
				{props.children}
			</div>
		</Placeholder>
	);
};

BlockPlaceholder.propTypes = {
	children: PropTypes.element.isRequired
};

/**
 * Form chooser select control
 */
export const ChooseForm = ({ formId, forms, onChange, labelText }) => {
	const { asSelectOptions } = useForms({ forms });
	const options = React.useMemo(() => {
		let opts = asSelectOptions();
		return [...[{ label: "---" }], ...opts];
	}, [forms, asSelectOptions]);
	return (
		<SelectControl
			label={labelText ? labelText : "Select Form"}
			value={formId}
			onChange={onChange}
			options={options}
		/>
	);
};

ChooseForm.propTypes = {
	formId: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
	forms: PropTypes.oneOfType([PropTypes.array, PropTypes.object]).isRequired,
	onChange: PropTypes.func.isRequired,
	labelText: PropTypes.string
};

/**
 * Ninja Forms block UI
 */
export default function Edit({
	formId,
	forms,
	labelText,
	setAttributes,
	siteUrl,
	previewToken
}) {
	const { getFormTitle } = useForms({ forms });
	const updateChosenForm = formId => {
		const formTitle = formId ? getFormTitle(formId) : "";
		setAttributes({
			formId: formId,
			formTitle
		});
	};

	if (!formId) {
		return (
			<BlockPlaceholder>
				<ChooseForm
					label={labelText ? labelText : "Select Form"}
					formId={formId}
					forms={forms}
					onChange={updateChosenForm}
					labelText={labelText}
				/>
			</BlockPlaceholder>
		);
	}

	return (
		<React.Fragment>
			<InspectorControls>
				<ChooseForm
					formId={formId}
					forms={forms}
					onChange={updateChosenForm}
					labelText={labelText}
				/>
			</InspectorControls>
			<FormPreviewIFrame
				siteUrl={siteUrl}
				previewToken={previewToken}
				formId={formId}
			/>
		</React.Fragment>
	);
}

Edit.propTypes = {
	formId: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
	forms: PropTypes.oneOfType([PropTypes.array, PropTypes.object]).isRequired,
	labelText: PropTypes.string,
	setAttributes: PropTypes.func.isRequired,
	siteUrl: PropTypes.string.isRequired,
	previewToken: PropTypes.string.isRequired
};
