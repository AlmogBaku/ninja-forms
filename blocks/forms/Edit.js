import React from 'react';

export default  function Edit() {
    return <React.Fragment/>
}