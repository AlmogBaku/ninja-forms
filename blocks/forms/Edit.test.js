import Edit from './Edit';
import {render,fireEvent} from '@testing-library/react';
describe( 'Form block Edit callback', () => {

    it( 'Matches snapshot', () => {
        const onChange = jest.fn();

        const {container} = render(<Edit onChange={onChange} form={'1'}/> );
        expect(container).toMatchSnapshot();
    });

    it( 'Calls onChange when changing form', () =>{
        const onChange = jest.fn();
        const {getByLabel} = render(<Edit onChange={onChange} form={'1'} /> );
        fireEvent.change(getByLabel('Form'), {
            target: {value: '3'}
        });
        expect(onChange).toBeCalledTimes(1);
        expect(onChange).toBeCalledWith('3')
    });
});