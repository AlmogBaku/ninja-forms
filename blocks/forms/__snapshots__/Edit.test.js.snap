// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form block Edit callback Matches snapshot 1`] = `<div />`;
